sdk/python/cmd/pulumi-language-python/main.go

```diff
@@ -960,5 +959,62 @@ func (host *pythonLanguageHost) GetProgramDependencies(
 func (host *pythonLanguageHost) RunPlugin(
 	req *pulumirpc.RunPluginRequest, server pulumirpc.LanguageRuntime_RunPluginServer,
 ) error {
-	return errors.New("not supported")
+	logging.V(5).Infof("Attempting to run python plugin in %s", req.Program)
+
+	args := []string{req.Program}
+	args = append(args, req.Args...)
+
+	var cmd *exec.Cmd
+	var err error
+	var virtualenv string
+	if host.virtualenv != "" {
+		virtualenv = host.virtualenvPath
+		if !python.IsVirtualEnv(virtualenv) {
+			return python.NewVirtualEnvError(host.virtualenv, virtualenv)
+		}
+		cmd = python.VirtualEnvCommand(virtualenv, "python", args...)
+	} else {
+		cmd, err = python.Command(server.Context(), args...)
+		if err != nil {
+			return err
+		}
+	}
+
+	closer, stdout, stderr, err := rpcutil.MakeRunPluginStreams(server, false)
+	if err != nil {
+		return err
+	}
+	// best effort close, but we try an explicit close and error check at the end as well
+	defer closer.Close()
+
+	cmd.Dir = req.Pwd
+	cmd.Env = req.Env
+	cmd.Stdout, cmd.Stderr = stdout, stderr
+
+	if virtualenv != "" {
+		cmd.Env = python.ActivateVirtualEnv(cmd.Env, virtualenv)
+	}
+	if err = cmd.Run(); err != nil {
+		if exiterr, ok := err.(*exec.ExitError); ok {
```

Comment:
This should probably be an errors.As.
The cast will probably continue to work for the foreseeable future, but it's not required to.

```suggestion
		if exiterr := &exec.ExitError{}; errors.As(err, &exiterr) {
```